### PR TITLE
2048: Prevent IA from showing on mobile devices

### PR DESCRIPTION
**Pull request by moollaza**
## Description of new Instant Answer, or changes
- Call `DDH.failed()` (before `onShow`) to allow Goodie to gracefully fail when JS loads on a mobile device

**Note:** This relies on internal changes to work correctly 👍 

## People to notify
@pjhampton 

## Testing & Review
To be completed by Community Leader (or DDG Staff) when reviewing Pull Request

**Pull Request**
- [ ] Title follows correct format (Specifies Instant Answer + Purpose)
- [ ] Description contains a valid Instant Answer Page Link (e.g. https://duck.co/ia/view/my_ia)

**Instant Answer Page** (for new Instant Answers)
- [ ] Instant Answer page is correctly filled out and contains:
    - The **Title** is appropriately named and formatted
    - The **IA topics** are present and appropriate
    - The **Description** is clear and coherent
    - **Source Name** exists if applicable
    - All **Example Queries** trigger on [Beta](https://beta.duckduckgo.com/)

**Code**
- [ ] Adheres to the [DuckDuckGo Style Guide](https://docs.duckduckhack.com/resources/code-style-guide.html)
- [ ] Behaviour is appropriately tested. If improvement, tests are adequately extended.
- [ ] There is no unnecessary files in place (such as editor config files)
- [ ] There is no API keys / secrets present
- [ ] Tests are passing (run `$ duckpan test <goodie_id>`)
    - Tester should report any failures

**Ready to merge?**

- [ ] Has this IA been deployed to and tested on [beta.duckduckgo.com](https://beta.duckduckgo.com/)?
- [ ] For larger commits, has this been approved be more than one community member?
- [ ] The number of reviews is appropriate for this type of PR
- [ ] The commit message is clear, coherent and fitting

**Pull Request Review Guidelines**: https://docs.duckduckhack.com/programming-mission/pr-review.html

---

Instant Answer Page: https://duck.co/ia/view/game2048

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: share/goodie/game2048/game2048.js [hunk 1]
```diff
@@ -2,7 +2,12 @@ DDH.game2048 = DDH.game2048 || {};
 
 DDH.game2048.build = function(ops) {
     "use strict";
-    
+
+    //Hide this goodie on mobile devices for now
+    if (DDG.device.isMobile || DDG.device.isMobileDevice) {
```
Comment (pjhampton) [inline share/goodie/game2048/game2048.js]: so moving this outside of onShow will kill it on mobile? Almost seems too good to be true haha

**Comment by pjhampton** [comment]
Oh, you said internal changes. Sure, I can wait for those.
Once those have been done I can add the CSS fixes and triggers

**Comment by moollaza** [comment]
The internal changes have been released. We should be able to test this now 👍

**Comment by moollaza** [comment]
Deploying to Beta...

**Comment by pjhampton** [comment]
Is there an issue with beta atm? It seems to be failing a lot

File: share/goodie/game2048/game2048.js (hunk 1 above)
Comment (moollaza) [inline share/goodie/game2048/game2048.js]: Yup 👍

**Comment by moollaza** [comment]
Deploy was failing because the IA page URL was wrong ☹️ 

All fixed now though 👍

**Comment by moollaza** [comment]
Whoops, this isn't going to work on Beta until it's updated -- will get that fixed soon 👍

**Pull request closed without merging**